catalog/content/core/content-core-catalogerplugin/src/test/java/ddf/content/plugin/cataloger/TestCatalogContentPlugin.java

```diff
@@ -0,0 +1,498 @@
+import ddf.content.operation.DeleteResponse;
+import ddf.content.operation.UpdateRequest;
+import ddf.content.operation.UpdateResponse;
+import ddf.content.plugin.ContentPlugin;
+import ddf.content.plugin.PluginExecutionException;
+import ddf.mime.MimeTypeToTransformerMapper;
+import ddf.security.SecurityConstants;
+import ddf.security.Subject;
+
+public class TestCatalogContentPlugin {
+
+    private static final String DEFAULT_METACARD_TRANSFORMER = "geojson";
+    private static final String MIME_BASE = "text";
+    private static final String MIME_SUB = "plain";
+    private static final String MIME_TYPE = MIME_BASE + "/" + MIME_SUB;
+    private static final String TEST_USER_NAME = "TestUser";
+    private static final String TEST_REALM = "testrealm";
+    private static final String TEST_DATA = "Test String From InputStream";
+    private static final String TEST_URI = "http://localhost/test.jpg";
+    private static final String BAD_URI = "bad uri";
+
+    private CatalogFramework mockCatalogFramework = mock(CatalogFramework.class);
+    private MimeTypeToTransformerMapper mockMimeTypeToTransformerMapper = mock(MimeTypeToTransformerMapper.class);
+    private BinaryContent mockBinaryContent = mock(BinaryContent.class);
+    private CreateRequest mockCreateRequest = mock(CreateRequest.class);
+    private UpdateRequest mockUpdateRequest = mock(UpdateRequest.class);
+    private CreateResponse mockInputCreateResponse = mock(CreateResponse.class);
+    private UpdateResponse mockInputUpdateResponse = mock(UpdateResponse.class);
+    private DeleteResponse mockInputDeleteResponse = mock(DeleteResponse.class);
+    private ContentItem mockContentItem = mock(ContentItem.class);
+    private MimeType mockMimeType = mock(MimeType.class);
+    private InputTransformer mockInputTransformer = mock(InputTransformer.class);
+    private ddf.catalog.operation.CreateResponse mockTestCreateResponse = mock(ddf.catalog.operation.CreateResponse.class);
+    private ddf.catalog.operation.UpdateResponse mockTestUpdateResponse = mock(ddf.catalog.operation.UpdateResponse.class);
+    private ddf.catalog.operation.DeleteResponse mockTestDeleteResponse = mock(ddf.catalog.operation.DeleteResponse.class);
+    private Metacard returnCard;
+    private List<Metacard> metacardList = new ArrayList<>();
+    private List<Update> updateList = new ArrayList<>();
+    private CatalogContentPlugin catalogContentPlugin = null;
+    private MetacardFactory metacardFactory = null;
+
+    @Before
+    public void setUp() throws Exception {
+        metacardFactory = new MetacardFactoryImpl();
+        catalogContentPlugin = new CatalogContentPlugin(mockCatalogFramework,
+                mockMimeTypeToTransformerMapper, metacardFactory);
+
+        setupContent();
+        setupMockContent();
+    }
+
+    @Test
+    public void setMetacardFactory() {
+        catalogContentPlugin.setMetacardFactory(metacardFactory);
+        assertNotNull(catalogContentPlugin);
```

Comment:
Generally, we prefer to use `assertThat()` over the other various `assert` methods. See [this page](http://junit.sourceforge.net/doc/ReleaseNotes4.4.html) for some of the advantages of `assertThat()`.